community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexStoreView.java

```diff
@@ -41,26 +42,39 @@
      *
      * @param labelIds array of label ids to generate updates for. Empty array means all.
      * @param propertyKeyIdFilter property key ids to generate updates for.
-     * @param propertyUpdateVisitor visitor which will see all generated {@link NodeUpdates}.
+     * @param propertyUpdateVisitor visitor which will see all generated {@link EntityUpdates}.
      * @param labelUpdateVisitor visitor which will see all generated {@link NodeLabelUpdate}.
      * @param forceStoreScan overrides decision about which source to scan from. If {@code true}
      * then store scan will be used, otherwise if {@code false} then the best suited will be used.
      * @return a {@link StoreScan} to start and to stop the scan.
      */
     <FAILURE extends Exception> StoreScan<FAILURE> visitNodes(
             int[] labelIds, IntPredicate propertyKeyIdFilter,
-            Visitor<NodeUpdates, FAILURE> propertyUpdateVisitor,
+            Visitor<EntityUpdates, FAILURE> propertyUpdateVisitor,
             Visitor<NodeLabelUpdate, FAILURE> labelUpdateVisitor,
             boolean forceStoreScan );
 
     /**
-     * Produces {@link NodeUpdates} objects from reading node {@code nodeId}, its labels and properties
+     * Retrieve all relationships in the database which has any of the the given relationship types AND
+     * one or more of the given property key ids.
+     *
+     * @param relationshipTypeIds array of relationsip type ids to generate updates for. Empty array means all.
+     * @param propertyKeyIdFilter property key ids to generate updates for.
+     * @param propertyUpdateVisitor visitor which will see all generated {@link EntityUpdates}
+     * @return a {@link StoreScan} to start and to stop the scan.
+     */
+    <FAILURE extends Exception> StoreScan<FAILURE> visitRelationships( int[] relationshipTypeIds, IntPredicate propertyKeyIdFilter,
+            Visitor<EntityUpdates,FAILURE> propertyUpdateVisitor );
+
+    /**
+     * Produces {@link EntityUpdates} objects from reading node {@code nodeId}, its labels and properties
      * and puts those updates into node updates container.
      *
      * @param nodeId id of node to load.
      * @return node updates container
      */
-    NodeUpdates nodeAsUpdates( long nodeId );
+    //TODO This seems to only be used in tests
```

Comment:
TODO again

Reply:
Right, this todo here... It only seems to be used in tests, but as far as I can tell, it has been that way for quite a while, and since we did not cause it, only noticed it, we did not prioritize fixing it. Would you rather see that we took time to remove it, removed the todo, or neither?